resdk/tests/functional/duplication/e2e_duplication.py

```diff
@@ -0,0 +1,39 @@
+# pylint: disable=missing-docstring
+from resdk.exceptions import ResolweServerError
+from resdk.tests.functional.base import BaseResdkFunctionalTest
+
+
+class TestDuplication(BaseResdkFunctionalTest):
+
+    def test_collection_duplication(self):
+        collection = self.res.collection.create(name='Test collection')
+        duplicate = collection.duplicate()
+        self.assertEqual(duplicate.name, 'Copy of Test collection')
+
+    def test_sample_duplication(self):
+        sample = self.res.sample.create(name='Test sample')
+        duplicate = sample.duplicate()
+        self.assertEqual(duplicate.name, 'Copy of Test sample')
+
+    def test_sample_duplication_perms(self):
+        collection = self.user_res.collection.create()
+        sample = self.user_res.sample.create(name='Test sample')
+        collection.add_samples(sample)
+
+        collection.permissions.remove_user(self.user_res.auth.username, 'add')
+        # TODO is it ok that no exception is raised when duplicating a sample
+        # with 'inherit_collection' set to True while the user has no 'add' permission
+        # on the collection?
+        # What happens is that the sample is duplicated just fine, but is also NOT
+        # added to the collection (without an exception being raised).
```

Comment:
This should be fixed in Resolwe. @hadalin can you do this?

Reply:
I could, but it's not a priority :)

I'd merge as it is and deal with this later as this is a separate issue.

Reply:
You can, but it doesn't make sense to add a test that works against expectations :)

Reply:
Well, these _are_ the current expectations :)

Either we

- [ ] keep the test which will fail once the issue is fixed in Resolwe, but we get to keep the test and not forget about it
- [ ] remove the test